### PR TITLE
faq: Add assurance case

### PR DESCRIPTION
**Pull request by kingthorin**
As part of working towards Silver on https://bestpractices.coreinfrastructure.org. This is the last item in the "Security" section.

This is just my initial thoughts on the doc, I'm happy to add/clarify more, just comment as you see fit.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: site/content/faq/what-is-zaps-assurance-case.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+ZAP is a Java application which is meant to be used primarily in Single user and CI/CD deployment models. As such protection of ZAP and it's associated/generated data is outside the scope of ZAP's sphere of influence (ex: if a system which houses ZAP can be physically stolen, then all the data on the system is at risk).
+
+While attempts are made to ensure that third parties cannot adversely impact users of ZAP, users should always be aware of what their targets are and what data or services ZAP may expose. To that end:
+
+- Users identify which "targets" they interact with.
+- The Web API and proxy are available only locally by default.
+- Access to the Web API requires the use of a key/token and identified permitted clients by default.
+- Scripts and add-ons are added based on user selection/choice.
+	- Scripts and add-ons which are maintained by the core team or advertised via the ZAP Marketplace are reviewed prior to release, to ensure at the very least that they do not do anything obviously malicious.
```
Comment (thc202) [inline site/content/faq/what-is-zaps-assurance-case.md:18]: Is the indentation on purpose?

File: site/content/faq/what-is-zaps-assurance-case.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Secure Design Principles
+
+All contributions to ZAP are reviewed by a minimum of two core team members. Design principals are further outlined in the [Developer Guide](https://www.zaproxy.org/docs/developer/) and [ZAP Wiki](https://github.com/zaproxy/zaproxy/wiki/).
+
+Known or intentional weaknesses such as allowing weak TLS/SSL connections or weak/improperly signed certificates are a side effect of the type of testing ZAP is designed to carry out, and exist simply to maximize coverage and computability with then potentially huge 'pool' of web enabled targets.
```
Comment (thc202) [inline site/content/faq/what-is-zaps-assurance-case.md:24]: - `compatibility`?
 - `then` → `the`?

File: site/content/faq/what-is-zaps-assurance-case.md (hunk 1 above)
Comment (kingthorin) [reply site/content/faq/what-is-zaps-assurance-case.md:18]: Yes, but I guess it doesn’t have to be 🤷‍♂️

**Comment by psiinon** [comment]
How about documenting some assumptions we make, like:

- We are not trying to protect from someone with command line access to the machine ZAP is running on
- We are aiming to protect users from the websites that ZAP connects to
- We do not (generally) trust websites that ZAP connects to
- We do trust specific ZAP websites, as long as we can connect to them over HTTPS
- If you are testing potentially malicious sites then we recommend taking additional precausions (as you should when dealing with malware), such as running ZAP in a container

(These are quick notes, not the suggested final wording)

**Comment by kingthorin** [comment]
Added.

**Comment by kingthorin** [comment]
@thc202 started to mention just at the end of the meeting, it could also go here: https://www.zaproxy.org/community/ in the "About" section I guess...

**Comment by kingthorin** [comment]
Ping team. Thoughts? Additions? Changes?

File: site/content/faq/what-is-zaps-assurance-case.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+weight: 7
+---
+
+## Basic Threat Model & Trust Boundaries
+
+ZAP is a Java application which is meant to be used primarily in Single user and CI/CD deployment models. As such protection of ZAP and it's associated/generated data is outside the scope of ZAP's sphere of influence (ex: if a system which houses ZAP can be physically stolen, then all the data on the system is at risk).
+
+The following assumptions are applicable for our model of ZAP's use:
+
+- We do not need to protect from someone with command line access to the machine ZAP is running on.
```
Comment (psiinon) [inline site/content/faq/what-is-zaps-assurance-case.md:14]: We do not "aim" ro protect.. ?

File: site/content/faq/what-is-zaps-assurance-case.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+- We aim to protect users from the websites that ZAP connects to.
+- We do not (generally) trust websites that ZAP connects to.
+- We do trust specific ZAP websites, as long as we can connect to them via HTTPS.
```
Comment (psiinon) [inline site/content/faq/what-is-zaps-assurance-case.md:17]: "and trust all of the signing certificates"

**Review by psiinon — COMMENTED** [review]
Just a couple of minor tweeks - feel free to reword or ignore

**Comment by kingthorin** [comment]
Thanks, will tweak that this evening.

**Comment by kingthorin** [comment]
Sorry for the delay, never fear, still on my radar :wink:

**Comment by kingthorin** [comment]
Tweaked.

**Review by psiinon — APPROVED** [review]

File: site/content/faq/what-is-zaps-assurance-case.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+### Common Implementation Security Weaknesses Countered
+
+In addition to the design principals mentioned above ZAP is subject to a [Bug Bounty](https://bugcrowd.com/owaspzap) program via BugCrowd. Issues identified via the BugBounty program are triaged and addressed appropriately. Examples of items tackled in the manner, thus far, include:
+- Installer DLL hijacking/side-jacking.
+- HTML Injection.
+- RCE and Info leaks via HUD.
+- API Key bypasses in shortcut handling.
+- XSS.
+
+Further ZAP's codebase is regularly tested via multiple Static Analyzers and issues addressed.
```
Comment (ricekot) [inline site/content/faq/what-is-zaps-assurance-case.md:43]:
```suggestion
Further, ZAP's codebase is regularly tested via multiple Static Analyzers and issues addressed.
```

File: site/content/faq/what-is-zaps-assurance-case.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
+---
+title: "What is ZAP's assurance case?"
+type: faq
+category: Detailed Questions
+weight: 7
+---
+
+## Basic Threat Model & Trust Boundaries
+
+ZAP is a Java application which is meant to be used primarily in Single user and CI/CD deployment models. As such protection of ZAP and it's associated/generated data is outside the scope of ZAP's sphere of influence (ex: if a system which houses ZAP can be physically stolen, then all the data on the system is at risk).
```
Comment (ricekot) [inline site/content/faq/what-is-zaps-assurance-case.md:10]:
```suggestion
ZAP is a Java application which is meant to be used primarily in Single user and CI/CD deployment models. As such protection of ZAP and its associated/generated data is outside the scope of ZAP's sphere of influence (ex: if a system which houses ZAP can be physically stolen, then all the data on the system is at risk).
```

Comment (kingthorin) [reply site/content/faq/what-is-zaps-assurance-case.md:10]: Good catch

**Comment by kingthorin** [comment]
Review addressed.

**Review by ricekot — APPROVED** [review]
Thanks!

**Comment by kingthorin** [comment]
@psiinon @thc202 good to go?

**Comment by psiinon** [comment]
Yep, I'm good 😁

File: site/content/faq/what-is-zaps-assurance-case.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+- We do not (generally) trust websites that ZAP connects to.
+- We do trust specific ZAP websites and all of the signing certificates, as long as we can connect to them via HTTPS.
+- If the user is testing potentially malicious sites we recommend taking additional precautions (as one would when dealing with malware), such as running ZAP in a container or VM.
+
+While attempts are made to ensure that third parties cannot adversely impact users of ZAP, users should always be aware of what their targets are and what data or services ZAP may expose. To that end:
+
+- Users identify which "targets" they interact with.
+- The Web API and proxy are available only locally by default.
+- Access to the Web API requires the use of a key/token and identified permitted clients by default.
+- Scripts and add-ons are added based on user selection/choice.
```
Comment (thc202) [inline site/content/faq/what-is-zaps-assurance-case.md:25]: Technically this is not true, we bundle add-ons without the user choosing them (more so with mandatory add-ons).

Comment (kingthorin) [reply site/content/faq/what-is-zaps-assurance-case.md:25]: Will tweak the text to be more specific.

**Pull request merged**